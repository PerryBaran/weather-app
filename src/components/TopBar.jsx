import React from "react";
import PropTypes from "prop-types";
import "../styles/TopBar.css";
import SearchForm from "./SearchForm";
import DropDownMenu from "./DropDownMenu";

function TopBar({ handleSearch }) {
  return (
    <div className="top-bar">
      <SearchForm handleSearch={handleSearch} />
      <DropDownMenu handleSearch={handleSearch} />
    </div>
  );
}

TopBar.propTypes = {
  handleSearch: PropTypes.func.isRequired,
};

export default TopBar;
