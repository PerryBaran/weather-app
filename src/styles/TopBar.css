.top-bar {
  display: flex;
  justify-content: space-between;
  width: 100vw;
  height: var(--top-bar-height);  
  padding: 10px 10px;
  border-bottom: var(--top-bar-border);
  background-color: var(--medium-blue);
}
