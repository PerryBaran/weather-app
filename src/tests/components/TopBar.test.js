import React from "react";
import { render } from "@testing-library/react";
import TopBar from "../../components/TopBar";

xdescribe("TopBar", () => {
  const validProps = {
    handleSearch: () => {},
    setSelectedUnits: () => {},
  };

  it("renders correctly", () => {
    const { asFragment } = render(
      <TopBar
        handleSearch={validProps.handleSearch}
        setSelectedUnits={validProps.setSelectedUnits}
      />
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
