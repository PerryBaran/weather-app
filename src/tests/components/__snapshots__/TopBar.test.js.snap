// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopBar snapshot 1`] = `
<DocumentFragment>
  <div
    class="top-bar"
  >
    <div
      class="search-form"
    >
      <input
        type="text"
      />
      <button
        type="submit"
      >
        Search
      </button>
    </div>
    <button
      class="drop-down-menu-button"
      type="button"
    >
      <img
        alt="hamburger menu"
        src="menu.png"
      />
    </button>
    <div
      class="drop-down-menu drop-down-menu__hidden"
      data-testid="drop-down-menu"
    >
      <form
        class="options-menu-form"
        name="options menu form"
      >
        <h2>
          Options
        </h2>
        <div
          class="options-menu-form__units"
        >
          <h3>
            Units:
          </h3>
          <label
            class="options-menu-form__label"
            for="standardRadio"
          >
            Scientific
            <input
              id="standardRadio"
              name="units"
              type="radio"
              value="standard"
            />
          </label>
          <label
            class="options-menu-form__label"
            for="metricRadio"
          >
            Metric
            <input
              checked=""
              id="metricRadio"
              name="units"
              type="radio"
              value="metric"
            />
          </label>
          <label
            class="options-menu-form__label"
            for="imperialRadio"
          >
            Imperial
            <input
              id="imperialRadio"
              name="units"
              type="radio"
              value="imperial"
            />
          </label>
        </div>
        <button
          class="options-menu-form__apply"
          type="submit"
        >
          Apply
        </button>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
